### ISSUE TITLE
Why I got an nan loss in training

### ISSUE THREAD

**Issue by Xuwanjun9** [THUNLP-AIPoet/WMPoetry#1]
I remove the .bool() method and use dtype=torch.uint8 instead because of the version of pytorch, but I got nan for loss, is there anyone meet the same problem?

**Comment by XiaoyuanYi** [comment]
@Xuwanjun9  Hi, can you provide more details， like the screen capture of your error information and the code lines you have changed， so that I can help you handle your problem？

**Comment by Xuwanjun9** [comment]
Thank you so much for your replying. 
The error information when I run the complete original code is like this:
![图片](https://user-images.githubusercontent.com/37858957/85817031-75aa4e80-b79f-11ea-8958-5284ac70a69f.png)
My torch version is 1.1.0, and there is no .bool() attribute in this version so I changed the code in graphs.py, line 168 and 360.
![图片](https://user-images.githubusercontent.com/37858957/85813275-4d1d5700-b795-11ea-8110-2eec58daa03a.png)，
![图片](https://user-images.githubusercontent.com/37858957/85813305-6920f880-b795-11ea-8ffb-f64f6cf2856a.png)
, where the original code should be 
![图片](https://user-images.githubusercontent.com/37858957/85813464-d7fe5180-b795-11ea-9e7a-b78c56937b3c.png) and 
![图片](https://user-images.githubusercontent.com/37858957/85813528-fbc19780-b795-11ea-88ba-13de5f71a05a.png). Except for this, I changed nothing.
Seems everything is fine when pre-training, the last epoch in pretraining is 
![图片](https://user-images.githubusercontent.com/37858957/85814966-d20a6f80-b799-11ea-8ffe-4d66bfdbbca1.png)
 .But the result of the changed code is like this in training process, 
 where for the first one, the ouput is empty but the loss is 9.658, however after several iteration, the output is always empty， and loss keeps nan. 
![图片](https://user-images.githubusercontent.com/37858957/85814909-adae9300-b799-11ea-91ed-48587ee6c4c4.png)
I‘ve tried to reduce the learing rate but it's unhelpful.
Thank you again for your help！

**Comment by Xuwanjun9** [comment]
> 
> 
> @Xuwanjun9 Hi, can you provide more details， like the screen capture of your error information and the code lines you have changed， so that I can help you handle your problem？

Thank you so much for your replying. 
The error information when I run the complete original code is like this:
![图片](https://user-images.githubusercontent.com/37858957/85817031-75aa4e80-b79f-11ea-8958-5284ac70a69f.png)
My torch version is 1.1.0, and there is no .bool() attribute in this version so I changed the code in graphs.py, line 168 and 360.
![图片](https://user-images.githubusercontent.com/37858957/85813275-4d1d5700-b795-11ea-8110-2eec58daa03a.png)，
![图片](https://user-images.githubusercontent.com/37858957/85813305-6920f880-b795-11ea-8ffb-f64f6cf2856a.png)
, where the original code should be 
![图片](https://user-images.githubusercontent.com/37858957/85813464-d7fe5180-b795-11ea-9e7a-b78c56937b3c.png) and 
![图片](https://user-images.githubusercontent.com/37858957/85813528-fbc19780-b795-11ea-88ba-13de5f71a05a.png). Except for this, I changed nothing.
Seems everything is fine when pre-training, the last epoch in pretraining is 
![图片](https://user-images.githubusercontent.com/37858957/85814966-d20a6f80-b799-11ea-8ffe-4d66bfdbbca1.png)
 .But the result of the changed code is like this in training process, 
 where for the first one, the ouput is empty but the loss is 9.658, however after several iteration, the output is always empty， and loss keeps nan. 
![图片](https://user-images.githubusercontent.com/37858957/85814909-adae9300-b799-11ea-91ed-48587ee6c4c4.png)
I‘ve tried to reduce the learing rate but it's unhelpful.
Thank you again for your help！

**Comment by Xuwanjun9** [comment]
> 
> 
> @Xuwanjun9 Hi, can you provide more details， like the screen capture of your error information and the code lines you have changed， so that I can help you handle your problem？

> 
> 
> @Xuwanjun9 Hi, can you provide more details， like the screen capture of your error information and the code lines you have changed， so that I can help you handle your problem？

Thank you so much for your replying. 
The error information when I run the complete original code is like this:
![图片](https://user-images.githubusercontent.com/37858957/85817031-75aa4e80-b79f-11ea-8958-5284ac70a69f.png)
My torch version is 1.1.0, and there is no .bool() attribute in this version so I changed the code in graphs.py, line 168 and 360.
![图片](https://user-images.githubusercontent.com/37858957/85813275-4d1d5700-b795-11ea-8110-2eec58daa03a.png)，
![图片](https://user-images.githubusercontent.com/37858957/85813305-6920f880-b795-11ea-8ffb-f64f6cf2856a.png)
, where the original code should be 
![图片](https://user-images.githubusercontent.com/37858957/85813464-d7fe5180-b795-11ea-9e7a-b78c56937b3c.png) and 
![图片](https://user-images.githubusercontent.com/37858957/85813528-fbc19780-b795-11ea-88ba-13de5f71a05a.png). Except for this, I changed nothing.
Seems everything is fine when pre-training, the last epoch in pretraining is 
![图片](https://user-images.githubusercontent.com/37858957/85814966-d20a6f80-b799-11ea-8ffe-4d66bfdbbca1.png)
 .But the result of the changed code is like this in training process, 
 where for the first one, the ouput is empty but the loss is 9.658, however after several iteration, the output is always empty， and loss keeps nan. 
![图片](https://user-images.githubusercontent.com/37858957/85814909-adae9300-b799-11ea-91ed-48587ee6c4c4.png)
I‘ve tried to reduce the learing rate but it's unhelpful.
Thank you again for your help！
